app/lib/theme/widgets/NumberWidget.tsx

```diff
@@ -1,24 +1,37 @@
 import { WidgetProps } from "@rjsf/core";
 import NumberFormat from "react-number-format";
 
-export const MoneyWidget: React.FC<WidgetProps> = ({
+export const NumberWidget: React.FC<WidgetProps> = ({
   schema,
   id,
   disabled,
+  uiSchema,
+  value,
   label,
   onChange,
-  value,
 }) => {
+  // If we are using this widget to show numbers as money or percent, we can set `isMoney` or `isPercentage` to true in the uiSchema.
+  const isMoney = uiSchema?.isMoney;
+  const isPercentage = uiSchema?.isPercentage;
+
+  // If we need to set the amount of decimal places, we can set it in the uiSchema, otherwise there will be no decimal places.
+  const numberOfDecimalPlaces = uiSchema.numberOfDecimalPlaces
+    ? uiSchema.numberOfDecimalPlaces
+    : isMoney
+    ? 2
+    : 0;
```

Comment:
In the diff component, we enforce money to have 2 decimal places, whereas here we allow the user to configure it. I'd be in favour of defaulting to 2 decimal places for money no matter what